### DISCUSSION TITLE
Why do we need to design the relative position in the group like this?

### DISCUSSION THREAD

**Discussion by chinyifei** [fabricjs/fabric.js#10024] [General]
If the coordinate position of a group member has a negative value, it will expand the entire group and cause the positions of other members to shift. This is different from the positioning logic in the canvas.
I have discovered many inconsistencies:the clipPath is positioned starting from the center of the object. The object's originX and originY values do not play any role, while clipPath originX and originY do. This is the same positioning logic used for fabric.Group.
Below is a demo I wrote. You can try changing the line coordinates.
https://stackblitz.com/edit/vitejs-vite-dkkg9y?file=src%2Fcomponents%2FHelloWorld.vue
<img width="675" alt="image" src="https://github.com/user-attachments/assets/a5f62d9f-3378-4ec3-9f2e-08550a7e76b5">

**Comment by asturur** [comment]
Yes there are many quirks. One of the changes will be removing all the origins options. Eventually long term also the canvas will have a default viewport with 0,0 set in the center
